lib/utils/tls.go

```diff
@@ -54,6 +62,74 @@ type CertificateName struct {
 	OrganizationalUnit []string `json:"org_unit"`
 }
 
+// TLSCredentials keeps the typical 3 components of a proper HTTPS configuration
+type TLSCredentials struct {
+	// PublicKey in PEM format
+	PublicKey []byte
+	// PrivateKey in PEM format
+	PrivateKey []byte
+	Cert       []byte
+}
+
+// GenerateSelfSignedCert generates a self signed certificate that
+// is valid for given domain names and ips, returns PEM-encoded bytes with key and cert
```

Comment:
Let's mention in the godoc that this method is supposed to generate certificate compatible with macOS requirements and insert a link to that Apple's KB article.